.github/workflows/ci.yml

```diff
@@ -0,0 +1,25 @@
+name: Run CI Suite
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  pytest:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.7"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pipenv
+      - name: Test with pytest
+        run: |
+          pipenv install --dev
```

Comment:
I'd have moved this up to the "Install dependencies" step, but it's not worth delaying the PR